requirements-dev.txt

```diff
@@ -20,5 +20,5 @@ jax; sys_platform == 'darwin' or sys_platform == 'linux'
 jaxlib; sys_platform == 'darwin' or sys_platform == 'linux'
 
 # Tensorflow does not yet have a 3.11 version
```

Comment:
I don't think this is true anymore. https://pypi.org/project/tensorflow/#files has 3.9-3.11

But fine to deal with in a dedicated PR to avoid a can of worms.

Reply:
Trying to update this here - since this PR is about updating python versions I think it's fine to handle it here.